### DISCUSSION TITLE
Cannot login to RDS environment on thin client

### DISCUSSION THREAD

**Discussion by MarcdeBorchgrave** [FreeRDP/FreeRDP#7394] [Q&A]
Hi all

We have an RDS environment (RD web, broker, file and host server) setup running Windows server 2019. We have Intel NUC's that are running Stratodesk and check into a central controller.

We have setup a second environment that we want our users to remote into. All settings and configs are the exact same as our first environment which works fine. When logging into the environment from a laptop using the RDP icon downloaded from the RD web server, it works fine. No errors at all. However when the exact same user tries to login to this environment from a thin client, they get the below error

[INFO][com.freerdp.core] - ERRINFO_SERVER_INSUFFICIENT_PRIVILEGES (0x00000009):The user cannot connect to the server due to insufficient access privileges.

I logged this to Stratodesk who said it's best I log it here. We are using Free RDP 1.2. Does anybody know what the issue could be? Any help would be much appreciated.

Regards

**Comment by akallabeth** [comment]
1. `FreeRDP 1.2` is old and we do no longer support that for quite some time now, please upgrade. (For comparison, debian `oldstable` has `2.0` already)
2. You´re using the same application, e.g. `xfreerdp` both times? Then you´re missing some command line option, but can´t tell without some details.

**Comment by MarcdeBorchgrave** [comment]
Hi Akallabeth

Apologies for only responding now. I forwarded your queries onto Stratodesk and they came back with the following

1. we do have 2.0 already included, 1.2 and 2.0 are the same version. The current version that is included is 2.0-8dc782d009, the version hash at the end shows exactly which commit was used for the build.
2. Yes we are using the same application. I was able to resolve the login issue by putting the following "/load-balance-info:"tsv://MS Terminal Services Plugin.1.Alpha" into User-defined Parameters. This allows a standard user to login to the environment fine.
3. However once logged in on the Intel NUC thin client, Once logged in, the session is extremely laggy and the start menu does not function correctly. Once again this only happens when logging in via a thin client. If I login from a laptop using the RD icon downloaded from the broker, I don't have this issue. I once again logged this to Stratodesk who suggested using the custom parameters of /gdi:hw and /gfx:RFX. I tried both of these to no effect. Stratodesk also suggested that you might have some other switches to improve graphics performance.

Can you advise?

Regards

**Reply by akallabeth** [reply]
@MarcdeBorchgrave ok,

1. `8dc782d009` is not based on `2.0` but `master` e.g. upcoming `3.0`
2. no, I only suggest `/network:auto` as all the other options commonly found on the internet do not work the way the users expect them to.

**Comment by MarcdeBorchgrave** [comment]
Hi Akallabeth

I have tried adding /network:auto into User-defined Parameters but the experience is still the same. Is this the correct way it should be?

/load-balance-info:"tsv://MS Terminal Services Plugin.1.Alpha" /network:auto

Is there anything else I can try to resolve this graphical issues I am experiencing.

Regards

**Reply by akallabeth** [reply]
only thing I can think of is trying the `rdp` file on the thin client too, maybe some option is missing and that is the easiest way to test (assuming you can somehow start the client with a `rdp` file?)
also, I don´t know how fast the devices are, so you might want to monitor the `CPU` usage during the session.
